crates/lambda-platform/src/winit/mod.rs

```diff
@@ -66,6 +78,77 @@ pub struct WindowHandle {
   pub monitor_handle: MonitorHandle,
 }
 
+// Should we take the loop as a field right here? Probably a ref or something? IDK
+pub struct WindowHandleBuilder {
+  window_handle: Option<Window>,
+  size: Option<WindowSize>,
+  monitor_handle: Option<MonitorHandle>,
+}
+
+impl WindowHandleBuilder {
+  /// Instantiate an empty builder
+  pub fn new() -> Self {
+    return Self;
+  }
+
+  /// Set the window size for the WindowHandle
+  fn with_window_size(
+    &mut self,
+    window_size: [u32; 2],
+    scale_factor: f64,
+  ) -> self {
+    let logical: LogicalSize<u32> = window_size.into();
+    let physical: PhysicalSize<u32> = logical_size.to_physical(scale_factor);
+
+    let window_size = WindowSize {
+      width: window_size[0],
+      height: window_size[1],
+      logical,
+      physical,
+    };
+
+    self.size = Some(window_size);
+    return self;
+  }
+
+  /// Probably the function that'll be used the most
+  pub fn with_window_properties<E: 'static>(
+    &mut self,
```

Comment:
Instead of passing the builder by mutable reference, you can just move it as mutable with: `mut self`